### DISCUSSION TITLE
aac decoder sound issue

### DISCUSSION THREAD

**Discussion by mehrfeuer** [baresip/baresip#2538] [General]
I just compiled baresip with fdk-aac 2.0.2.
It‘s working smooth with aac-ld or aac-lc to a 2wcom hardware codec and even with luci software products.
When i try to call a prodys prontonet or aeta scoopy+s hardware codecs i always get noisy crackling sound on baresip output.

aac: aacDecoder_DecoderFrame()failed (0x4006) / (0x4002)
audio: mpeg4-generic codec decode 243 bytes: Protocol error [71]

I tried to adjust aac_aot and aac_constantduration in the config with no luck.

**Comment by sreimers** [comment]
Hard to debug without access to the hardware. Maybe you can compare with wireshark if there any protocol differences. For now I convert this issue to a discussion, if you can pinpoint the problem we can open a issue.

**Comment by premultiply** [comment]
I've seen that kind of issue with some of theses professional devices which send an fmtp description which does not match the properties of the aac bitstream they send. Or they do not adapt to the fmtp description they receive from baresip.
baresips aac implementaion relies on the fmtp metadata in sip session setup for initialization of its aac decoder and encoder.

I think AVT devices had the same bug as caller as far as I remember.

So this is a bug in the device firmware the vendor has to fix.

Maybe try to reverse the call direction as workaround.
